angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
I think the existing code is correct as is. It gives the correct types for people using strict null checks, but 'any' for those who aren't, which seems correct behaviour.

Note you can get the correct null return type when typing the paramater as null, even without strict null checks. e.g
[playground](http://www.typescriptlang.org/play/#src=interface%20OrderBy%20{%20%20%20%20<T%2C%20A%20extends%20T[]%20|%20null%20|%20undefined>(%20%20%20%20%20%20%20%20array%3A%20A%2C%20%20%20%20%20%20%20%20expression%3F%3A%20string%20|%20((value%3A%20T)%20%3D>%20any)%20|%20(((value%3A%20T)%20%3D>%20any)%20|%20string)[]%2C%20%20%20%20%20%20%20%20reverse%3F%3A%20boolean%20%20%20%20)%3A%20A%3B}let%20f%3A%20OrderBy%3Blet%20a%20%3D%20f(null%2C%20'z')%3B%20%2F%2F%20`a`%20is%20of%20type%20`any`%20not%20`null`)